cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java

```diff
@@ -286,17 +302,32 @@ public String cloneRemote()
     }
     // Clean up the directory if it already exists.
     deletePathIfExists(getRepositoryRoot());
-    RepositoryConfig repositoryConfig = sourceControlConfig
-        .getRepositoryConfig();
-    CloneCommand command =
-        createCommand(Git::cloneRepository).setURI(repositoryConfig.getLink())
-            .setDirectory(getRepositoryRoot().toFile());
+    RepositoryConfig repositoryConfig = sourceControlConfig.getRepositoryConfig();
+    CloneCommand command = createCommand(Git::cloneRepository).setURI(
+        repositoryConfig.getLink()).setDirectory(getRepositoryRoot().toFile());
     String branch = getBranchRefName(repositoryConfig.getDefaultBranch());
     if (branch != null) {
       command.setBranchesToClone(Collections.singleton((branch)))
           .setBranch(branch);
     }
+
+    final Stopwatch stopwatch = new Stopwatch().start();
     git = command.call();
+    final long cloneTimeMillis = stopwatch.stop()
+        .elapsedTime(TimeUnit.MILLISECONDS);
+
+    // Record the repository size metric.
+    try {
+      long repoSize = calculateDirectorySize(getRepositoryRoot());
+      metricsContext.event(SourceControlManagement.CLONE_REPOSITORY_SIZE_MB,
+          repoSize / MEGA_BYTES);
```

Comment:
Any specific reason you want to do the rounding? It also adds confusion because of MB meaning (1e6 vs 2^20)

Reply:
We don't expect many Git repositories to be less than 1 MB. The event method takes long (not double). 
I've changed this to use bytes now. We can do the scaling in the queries.